modules/wallet/transactions_test.go

```diff
@@ -9,75 +9,57 @@ import (
 // testFundTransaction funds and completes a transaction using the
 // build-your-own transaction functions, checking that a no-refund transaction
 // is created that is valid.
-func (wt *WalletTester) testFundTransaction() {
-	// Get a coin address for the wallet and fund the wallet using money from
-	// the
-	addr, _, err := wt.CoinAddress()
-	if err != nil {
-		wt.Fatal(err)
-	}
-	siacoinOutput, value := wt.FindSpendableSiacoinInput()
-	walletFunderTxn := wt.AddSiacoinInputToTransaction(consensus.Transaction{}, siacoinOutput)
-	walletFunderOutput := consensus.SiacoinOutput{
-		Value:      value,
-		UnlockHash: addr,
-	}
-	walletFunderTxn.SiacoinOutputs = append(walletFunderTxn.SiacoinOutputs, walletFunderOutput)
-	block := wt.MineCurrentBlock([]consensus.Transaction{walletFunderTxn})
-	err = wt.State.AcceptBlock(block)
-	if err != nil {
-		wt.Fatal(err)
-	}
-
+func (wt *walletTester) testFundTransaction() {
 	// Build a transaction that intentionally needs a refund.
-	id, err := wt.RegisterTransaction(consensus.Transaction{})
+	id, err := wt.wallet.RegisterTransaction(consensus.Transaction{})
+	fund := wt.wallet.Balance(false).Sub(consensus.NewCurrency64(1))
 	if err != nil {
-		wt.Fatal(err)
+		wt.t.Fatal(err)
 	}
-	_, err = wt.FundTransaction(id, value.Sub(consensus.NewCurrency64(1)))
+	_, err = wt.wallet.FundTransaction(id, fund)
 	if err != nil {
-		wt.Fatal(err)
+		wt.t.Fatal(err)
 	}
-	_, _, err = wt.AddMinerFee(id, value.Sub(consensus.NewCurrency64(1)))
+	wt.updateWait()
+	_, _, err = wt.wallet.AddMinerFee(id, fund)
 	if err != nil {
-		wt.Fatal(err)
+		wt.t.Fatal(err)
 	}
-	t, err := wt.SignTransaction(id, true)
+	t, err := wt.wallet.SignTransaction(id, true)
 	if err != nil {
-		wt.Fatal(err)
+		wt.t.Fatal(err)
 	}
 	err = wt.tpool.AcceptTransaction(t)
 	if err != nil {
-		wt.Fatal(err)
+		wt.t.Fatal(err)
 	}
+	wt.updateWait()
 
 	// Check that the length of the created transaction is 1 siacoin, and that
 	// the unconfirmed balance of the wallet is 1.
 	if len(t.SiacoinOutputs) != 0 {
-		wt.Error("more than zero siacoin outputs created in custom transaction")
+		wt.t.Error("more than zero siacoin outputs created in custom transaction")
 	}
-	if wt.Balance(true).Cmp(consensus.NewCurrency64(1)) != 0 {
-		wt.Error(wt.Balance(true).MarshalJSON())
-		wt.Error("wallet balance not reporting at one?")
+	if wt.wallet.Balance(true).Cmp(consensus.NewCurrency64(1)) != 0 {
+		wt.t.Error(wt.wallet.Balance(true).MarshalJSON())
```

Comment:
Does `Currency` still not have a `String` method? If it does, `MarshalJSON` isn't necessary here